### PR TITLE
Fix readAlias when Posix error

### PR DESCRIPTION
**Pull request by ChristopheLarchier**

### CODE REVIEW

**Review by luc-guyot-infomaniak — APPROVED** [review]
Only minor comments.

File: src/libcommonserver/io/iohelper_mac.mm [hunk 1]
```diff
@@ -46,13 +50,17 @@
...
+    }
+
     NSURL *pathURL = [NSURL fileURLWithPath:pathStr];
 
     NSError *error = nil;
     NSNumber *isAliasNumber = nil;
     BOOL ret = [pathURL getResourceValue:&isAliasNumber forKey:NSURLIsAliasFileKey error:&error];
     if (!ret) {
-        ioError = nsError2ioError(error.code);
+        ioError = nsError2ioError(error);
```
Comment (ClementKunz) [inline src/libcommonserver/io/iohelper_mac.mm:63]: Same, should we log?

File: src/libcommonserver/io/iohelper_mac.mm [hunk 2]
```diff
@@ -80,11 +88,12 @@
     CFRelease(bookmarkRef);
     CFRelease(aliasUrl);
     if (!ret) {
-        CFIndex idx = CFErrorGetCode(error);
-        CFRelease(error);
-        ioError = nsError2ioError(idx);
-        if (ioError != IoErrorUnknown) {
-            return true;
+        if (error) {
```
Comment (ClementKunz) [inline src/libcommonserver/io/iohelper_mac.mm:91]: same

File: src/libcommonserver/io/iohelper.cpp [hunk 3]
```diff
@@ -155,6 +155,11 @@ bool IoHelper::_isExpectedError(IoError ioError) noexcept {
   \return true if no error occurred or if the error is expected, false otherwise.
 */
 bool IoHelper::_setTargetType(ItemType &itemType) noexcept {
+    if (itemType.targetPath.empty()) {
+        itemType.targetType = NodeTypeUnknown;
+        return true;
```
Comment (ClementKunz) [inline src/libcommonserver/io/iohelper.cpp:160]: Should we really return true? An item with type `NodeTypeUnknown`is not a normal behavior

File: src/libcommonserver/io/iohelper_mac.mm [hunk 4]
```diff
@@ -80,11 +88,12 @@
...
+            ioError = nsError2ioError((NSError *)error);
+            CFRelease(error);
+            if (ioError != IoErrorUnknown) {
```
[the remark this answers is not in the document]
Comment (ChristopheLarchier) [reply src/libcommonserver/io/iohelper_mac.mm:103]: Code change making the remark obsolete.

File: src/libcommonserver/io/iohelper_mac.mm [hunk 5]
```diff
@@ -106,9 +115,8 @@
     CFRelease(aliasUrl);
     if (bookmarkRef == nil) {
         if (error) {
-            CFIndex idx = CFErrorGetCode(error);
+            ioError = nsError2ioError((NSError *)error);
             CFRelease(error);
-            ioError = nsError2ioError(idx);
             if (ioError != IoErrorUnknown) {
```
[the remark this answers is not in the document]
Comment (ChristopheLarchier) [reply src/libcommonserver/io/iohelper_mac.mm:134]: Code change making the remark obsolete.

File: src/libcommonserver/io/iohelper_mac.mm [hunk 6]
```diff
@@ -46,13 +50,17 @@
     ioError = IoErrorSuccess;
 
     NSString *pathStr = [NSString stringWithCString:path.c_str() encoding:NSUTF8StringEncoding];
+    if (pathStr == nil) {
+        return false;
```
[the remark this answers is not in the document]
Comment (ChristopheLarchier) [reply src/libcommonserver/io/iohelper_mac.mm:56]: Done

File: src/libcommonserver/io/iohelper_mac.mm (hunk 1 above)
Comment (ChristopheLarchier) [reply src/libcommonserver/io/iohelper_mac.mm]: Done

File: src/libcommonserver/io/iohelper_mac.mm (hunk 2 above)
Comment (ChristopheLarchier) [reply src/libcommonserver/io/iohelper_mac.mm:100]: Done

File: src/libcommonserver/io/iohelper.cpp (hunk 3 above)
Comment (ChristopheLarchier) [reply src/libcommonserver/io/iohelper.cpp:160]: This can happen when the target path of a link is empty (dangling link), this is not an error.

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Pull request merged**